extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/MessageBundleProcessor.java

```diff
@@ -342,8 +343,12 @@ private void validateExpression(BuildProducer<IncorrectExpressionBuildItem> inco
             return;
         }
         if (!expression.hasNamespace()) {
-            String name = expression.getParts().get(0).getName();
-            if (!paramNames.contains(name)) {
+            Expression.Part firstPart = expression.getParts().get(0);
+            String name = firstPart.getName();
+            // Skip expressions that have type info derived from a parent section, i.e. it<loop#3> nad foo<set#3>
```

Comment:
Shouldn't it be `// (...) it<loop#3> and foo<set#3>`?

Reply:
Yup :D

Reply:
Fixed ;-)